Crash when parsing pymssql._mssql

### Bug description

When parsing the following ``a.py``:

<!--
 If sharing the code is not an option, please state so,
 but providing only the stacktrace would still be helpful.
 -->

```python
from pymssql import _mssql

print(_mssql.VERSION)
```

### .pylintrc
```toml
[MASTER]

# A comma-separated list of package or module names from where C extensions may
# be loaded. Extensions are loading into the active Python interpreter and may
# run arbitrary code.
extension-pkg-whitelist=pymssql._mssql
```

### Command used

```shell
pylint --rcfile=.pylintrc a.py
```

### Pylint output

<details open>
    <summary>
        pylint crashed with a ``AstroidError`` and with the following stacktrace:
    </summary>

```python
Traceback (most recent call last):
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/lint/pylinter.py", line 824, in _lint_file
    check_astroid_module(module)
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/lint/pylinter.py", line 1056, in check_astroid_module
    retval = self._check_astroid_module(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/lint/pylinter.py", line 1108, in _check_astroid_module
    walker.walk(node)
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/utils/ast_walker.py", line 90, in walk
    self.walk(child)
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/utils/ast_walker.py", line 87, in walk
    callback(astroid)
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/checkers/variables.py", line 2171, in visit_importfrom
    self._check_module_attrs(node, module, name.split("."))
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/checkers/variables.py", line 3221, in _check_module_attrs
    module = module.getattr(name)[0]
             ^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 369, in getattr
    result = [self.import_module(name, relative_only=True)]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 463, in import_module
    return AstroidManager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/manager.py", line 241, in ast_from_module_name
    return self.ast_from_module(named_module, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/manager.py", line 340, in ast_from_module
    return AstroidBuilder(self).module_build(module, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/builder.py", line 109, in module_build
    node = self._manager.visit_transforms(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/manager.py", line 129, in visit_transforms
    return self._transform.visit(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/transforms.py", line 163, in visit
    return self._visit(node)
           ^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/transforms.py", line 85, in _visit
    visited = self._visit_generic(value)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/transforms.py", line 113, in _visit_generic
    return [self._visit_generic(child) for child in node]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/transforms.py", line 120, in _visit_generic
    return self._visit(node)
           ^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/transforms.py", line 88, in _visit
    return self._transform(node)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/transforms.py", line 67, in _transform
    if predicate is None or predicate(node):
                            ^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/brain/brain_namedtuple_enum.py", line 655, in _is_enum_subclass
    return cls.is_subtype_of("enum.Enum")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2015, in is_subtype_of
    return any(anc.qname() == type_name for anc in self.ancestors(context=context))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2015, in <genexpr>
    return any(anc.qname() == type_name for anc in self.ancestors(context=context))
                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2189, in ancestors
    for baseobj in stmt.infer(context):
                   ^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/node_ng.py", line 164, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/decorators.py", line 81, in inner
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/decorators.py", line 44, in wrapped
    for res in _func(node, context, **kwargs):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/node_classes.py", line 580, in _infer
    frame, stmts = self.lookup(self.name)
                   ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/_base_nodes.py", line 276, in lookup
    return self.scope().scope_lookup(self, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 2155, in scope_lookup
    return frame._scope_lookup(node, name, offset)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/scoped_nodes/mixin.py", line 83, in _scope_lookup
    stmts = _filter_stmts(node, self.locals[name], self, offset)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/filter_statements.py", line 112, in _filter_stmts
    statements = _get_filtered_node_statements(base_node, stmts)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/filter_statements.py", line 25, in _get_filtered_node_statements
    statements = [(node, node.statement()) for node in stmt_nodes]
                         ^^^^^^^^^^^^^^^^
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/astroid/nodes/scoped_nodes/scoped_nodes.py", line 414, in statement
    raise StatementMissing(target=self)
astroid.exceptions.StatementMissing: Statement not found on <Module.datetime l.0 at 0x7cae33553140>

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/lint/pylinter.py", line 787, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "/home/marko/Code/issue/lib64/python3.12/site-packages/pylint/lint/pylinter.py", line 826, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError

```


</details>

### Expected behavior

No crash.
If I run pylint _without_ `extension_pkg_whitelist` in the `pylintrc` file, then there is no crash, but the following warning is present in the report
```
pylint_astroid_pymssql.py:3:6: I1101: Module 'pymssql._mssql' has no 'VERSION' member, but source is unavailable. Consider adding this module to extension-pkg-allow-list if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
```

### Pylint version

```shell
pylint 4.0.6
astroid 4.0.4
Python 3.12.13 (main, Jul 10 2026, 00:00:00) [GCC 11.5.0 20240719 (Red Hat 11.5.0-14)]
```

### OS / Environment

linux (Linux)

### Additional dependencies
Only `pymssql` and `pylint` installed in a `python3.12` venv
```
Package      Version
------------ -------
astroid      4.0.4
dill         0.4.1
isort        8.0.1
mccabe       0.7.0
pip          26.1.2
platformdirs 4.10.0
pylint       4.0.6
pymssql      2.3.13
tomlkit      0.15.1
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -7,6 +7,11 @@ What's New in astroid 4.2.0?
 ============================
 Release date: TBA
 
+* Prevent crashes while processing extension-module classes whose bases cannot
+  be resolved when applying enum inference transformations.
+
+  Closes pylint-dev/pylint#11179
+
 * Comprehension conditions now constrain the inference of names used in the
   comprehension, like ``if`` statement conditions already did:
   ``[x for x in lst if x is not None]`` no longer infers ``None`` for ``x``

--- a/astroid/brain/brain_namedtuple_enum.py
+++ b/astroid/brain/brain_namedtuple_enum.py
@@ -16,6 +16,7 @@ from astroid import arguments, bases, nodes, util
 from astroid.builder import AstroidBuilder, _extract_single_node, extract_node
 from astroid.context import InferenceContext
 from astroid.exceptions import (
+    AstroidError,
     AstroidTypeError,
     AstroidValueError,
     InferenceError,
@@ -654,7 +655,10 @@ def _get_namedtuple_fields(node: nodes.Call) -> str:
 
 def _is_enum_subclass(cls: nodes.ClassDef) -> bool:
     """Return whether cls is a subclass of an Enum."""
-    return cls.is_subtype_of("enum.Enum")
+    try:
+        return cls.is_subtype_of("enum.Enum")
+    except AstroidError:
+        return False
 
 
 def register(manager: AstroidManager) -> None:

--- a/tests/brain/test_enum.py
+++ b/tests/brain/test_enum.py
@@ -4,6 +4,8 @@
 
 from __future__ import annotations
 
+import datetime
+import types
 import unittest
 
 import pytest
@@ -11,9 +13,23 @@ import pytest
 import astroid
 from astroid import bases, builder, nodes, objects, util
 from astroid.exceptions import InferenceError
+from astroid.manager import AstroidManager
 
 
 class EnumBrainTest(unittest.TestCase):
+    def test_enum_predicate_does_not_raise_for_extension_class(self) -> None:
+        """Classes introspected from extensions can have an unresolvable base.
+
+        Regression test for pylint-dev/pylint#11179.
+        """
+        module = types.ModuleType("extension_module")
+        module.datetime = datetime
+        extension_class = type("ExtensionClass", (datetime.datetime,), {})
+        extension_class.__module__ = module.__name__
+        module.ExtensionClass = extension_class
+
+        builder.AstroidBuilder(AstroidManager()).module_build(module, module.__name__)
+
     def test_simple_enum(self) -> None:
         module = builder.parse("""
         import enum

